File: align-ios/Align/Data/SessionStore.swift
import Foundation

/// Token sesiune Align (`x-session-token` / valoare `align_sid`) + userId. Produție: WebView bridge sau ASWebAuthenticationSession.
enum SessionStore {
    private static let ud = UserDefaults.standard
    private enum K {
        static let session = "align_session_token"
        static let userId = "align_user_id"
    }

    static var sessionToken: String? {
        get { ud.string(forKey: K.session) }
        set { ud.set(newValue, forKey: K.session) }
    }

    static var userId: String? {
        get { ud.string(forKey: K.userId) }
        set { ud.set(newValue, forKey: K.userId) }
    }

    static var isLoggedIn: Bool {
        guard let s = sessionToken, let u = userId else { return false }
        return !s.isEmpty && !u.isEmpty
    }
}

File: align-android/app/src/main/java/ro/align/app/data/SessionStore.kt
package ro.align.app.data

import android.content.Context

/**
 * Stochează `align_sid` echivalent: tokenul de sesiune și userId pentru API Align.
 * Produție: populat din WebView după login (JavaScript bridge) sau OAuth device flow.
 */
class SessionStore(context: Context) {
    private val p = context.applicationContext.getSharedPreferences(PREFS, Context.MODE_PRIVATE)

    var sessionToken: String?
        get() = p.getString(KEY_SESSION, null)
        set(v) {
            p.edit().putString(KEY_SESSION, v).apply()
        }

    var userId: String?
        get() = p.getString(KEY_USER_ID, null)
        set(v) {
            p.edit().putString(KEY_USER_ID, v).apply()
        }

    fun clear() {
        p.edit().remove(KEY_SESSION).remove(KEY_USER_ID).apply()
    }

    fun isLoggedIn(): Boolean = !sessionToken.isNullOrBlank() && !userId.isNullOrBlank()

    companion object {
        private const val PREFS = "align_session"
        private const val KEY_SESSION = "session_token"
        private const val KEY_USER_ID = "user_id"
    }
}
